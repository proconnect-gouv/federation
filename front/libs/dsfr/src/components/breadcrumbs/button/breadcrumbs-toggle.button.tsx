import { useToggle } from 'usehooks-ts';

import { t } from '@fc/i18n';

interface BreadCrumbsToggleButtonProps {
  id: string;
}

export const BreadCrumbsToggleButton = ({ id }: BreadCrumbsToggleButtonProps) => {
  const [show, toggleShow] = useToggle(false);

  return (
    <button
      aria-controls={id}
      aria-expanded={show}
      className="fr-breadcrumb__button"
      onClick={toggleShow}>
      {t('DSFR.breadcrumbs.show')}
    </button>
  );
};

BreadCrumbsToggleButton.displayName = 'BreadCrumbsToggleButton';
