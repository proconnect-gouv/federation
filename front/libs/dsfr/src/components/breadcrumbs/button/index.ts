export * from './breadcrumbs-toggle.button';
