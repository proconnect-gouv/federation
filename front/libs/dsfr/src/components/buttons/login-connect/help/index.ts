export * from './login-connect.help';
