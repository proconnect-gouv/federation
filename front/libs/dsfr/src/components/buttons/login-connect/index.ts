export * from './login-connect.component';
