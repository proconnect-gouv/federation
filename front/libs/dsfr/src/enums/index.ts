/* istanbul ignore file */

// declarative file
export * from './alert-types.enum';
export * from './button-types.enum';
export * from './icon-placement.enum';
export * from './pagination.enum';
export * from './priorities.enum';
export * from './sizes.enum';
