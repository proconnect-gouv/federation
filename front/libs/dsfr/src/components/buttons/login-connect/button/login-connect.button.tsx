import classnames from 'classnames';
import React from 'react';

import type { PropsWithOnClick } from '@fc/common';
import { t } from '@fc/i18n';

import type { ButtonTypes } from '../../../../enums';
import { ConnectTypes } from '../../../../enums';
import styles from './login-connect.button.module.scss';

interface LoginConnectButtonProps extends PropsWithOnClick {
  type: ButtonTypes;
  connectType: ConnectTypes;
}

export const LoginConnectButton = React.memo(
  ({ connectType, onClick, type }: LoginConnectButtonProps) => {
    const loginLabel = t('DSFR.button.loginConnect.authenticateWith');
    return (
      <button
        className={classnames('fr-connect', {
          [styles['fr-connect--proconnect']]: connectType === ConnectTypes.PRO_CONNECT,
        })}
        type={type}
        onClick={onClick}>
        <span className="fr-connect__login">{loginLabel}</span>
        <span className="fr-connect__brand">{connectType}</span>
      </button>
    );
  },
);

LoginConnectButton.displayName = 'LoginConnectButton';
