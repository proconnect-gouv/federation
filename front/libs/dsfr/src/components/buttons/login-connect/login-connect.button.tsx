import classnames from 'classnames';
import React from 'react';

import { ButtonTypes, ConnectTypes } from '../../../enums';

interface LoginConnectButtonProps
  extends Pick<React.HTMLProps<HTMLButtonElement>, 'onClick' | 'className'> {
  showIcon?: boolean;
  showHelp?: boolean;
  type?: ButtonTypes;
  connectType: ConnectTypes;
}

export const LoginConnectButton = React.memo(
  ({
    className,
    connectType,
    onClick,
    showHelp = false,
    showIcon = false,
    type = ButtonTypes.BUTTON,
  }: LoginConnectButtonProps) => {
    const target = showIcon ? '_blank' : undefined;
    const rel = showIcon ? 'noreferrer' : undefined;
    const url =
      connectType === ConnectTypes.FRANCE_CONNECT
        ? 'https://franceconnect.gouv.fr/'
        : 'https://agentconnect.gouv.fr/';
    const label = `Qu’est ce que ${connectType} ?`;
    return (
      <div className={classnames('fr-connect-group', className)} data-testid="LoginConnectButton">
        <button className="fr-connect" type={type} onClick={onClick}>
          <span className="fr-connect__login">S’identifier avec</span>
          <span className="fr-connect__brand">{connectType}</span>
        </button>
        {showHelp && (
          <p>
            <a href={url} rel={rel} target={target} title={`${label} - nouvelle fenêtre`}>
              {label}
            </a>
          </p>
        )}
      </div>
    );
  },
);

LoginConnectButton.displayName = 'LoginConnectButton';
