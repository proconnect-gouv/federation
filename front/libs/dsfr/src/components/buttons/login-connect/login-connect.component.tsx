import classnames from 'classnames';
import React from 'react';

import type { PropsWithClassName, PropsWithOnClick } from '@fc/common';

import type { ConnectTypes } from '../../../enums';
import { ButtonTypes } from '../../../enums';
import { LoginConnectButton } from './button';
import { LoginConnectHelp } from './help';

interface LoginConnectComponentProps extends PropsWithClassName, PropsWithOnClick {
  showIcon?: boolean;
  showHelp?: boolean;
  type?: ButtonTypes;
  connectType: ConnectTypes;
}

export const LoginConnectComponent = React.memo(
  ({
    className,
    connectType,
    onClick,
    showHelp = false,
    showIcon = false,
    type = ButtonTypes.BUTTON,
  }: LoginConnectComponentProps) => (
    <div className={classnames('fr-connect-group', className)} data-testid="LoginConnectComponent">
      <LoginConnectButton connectType={connectType} type={type} onClick={onClick} />
      {showHelp && <LoginConnectHelp connectType={connectType} showIcon={showIcon} />}
    </div>
  ),
);

LoginConnectComponent.displayName = 'LoginConnectComponent';
