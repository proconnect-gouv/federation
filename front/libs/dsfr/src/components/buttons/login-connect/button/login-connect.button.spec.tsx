import { render } from '@testing-library/react';

import { t } from '@fc/i18n';

import { ButtonTypes, ConnectTypes } from '../../../../enums';
import { LoginConnectButton } from './login-connect.button';

describe('LoginConnectButton', () => {
  beforeEach(() => {
    // Given
    jest.mocked(t).mockReturnValue('any-autheticate-i81n-mock');
  });

  it('shoud match snapshot, with connect type as FranceConnect', () => {
    // Given
    // When
    const { container, getByText } = render(
      <LoginConnectButton connectType={ConnectTypes.FRANCE_CONNECT} type={ButtonTypes.BUTTON} />,
    );
    const connectLabelElt = getByText('FranceConnect');
    const authenticateLabelElt = getByText('any-autheticate-i81n-mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveAttribute('type', 'button');
    expect(container.firstChild).toHaveClass('fr-connect');
    expect(connectLabelElt).toBeInTheDocument();
    expect(authenticateLabelElt).toBeInTheDocument();
    expect(t).toHaveBeenCalledOnce();
    expect(t).toHaveBeenCalledWith('DSFR.button.loginConnect.authenticateWith');
  });

  it('shoud match snapshot, with connect type as ProConnect', () => {
    // Given
    // When
    const { container, getByText } = render(
      <LoginConnectButton connectType={ConnectTypes.PRO_CONNECT} type={ButtonTypes.BUTTON} />,
    );
    const connectLabelElt = getByText('ProConnect');
    const authenticateLabelElt = getByText('any-autheticate-i81n-mock');

    // Then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveAttribute('type', 'button');
    expect(container.firstChild).toHaveClass('fr-connect fr-connect--proconnect');
    expect(connectLabelElt).toBeInTheDocument();
    expect(authenticateLabelElt).toBeInTheDocument();
    expect(t).toHaveBeenCalledOnce();
    expect(t).toHaveBeenCalledWith('DSFR.button.loginConnect.authenticateWith');
  });

  it('shoud match snapshot, with a defined button type', () => {
    // When
    const { container } = render(
      <LoginConnectButton connectType={ConnectTypes.PRO_CONNECT} type={ButtonTypes.RESET} />,
    );

    // Then
    expect(container).toMatchSnapshot();
    expect(container.firstChild).toHaveAttribute('type', 'reset');
  });
});
