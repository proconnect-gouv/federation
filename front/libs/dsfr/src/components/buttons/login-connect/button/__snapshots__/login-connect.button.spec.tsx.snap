// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginConnectButton shoud match snapshot, with a defined button type 1`] = `
<div>
  <button
    class="fr-connect fr-connect--proconnect"
    type="reset"
  >
    <span
      class="fr-connect__login"
    >
      any-autheticate-i81n-mock
    </span>
    <span
      class="fr-connect__brand"
    >
      ProConnect
    </span>
  </button>
</div>
`;

exports[`LoginConnectButton shoud match snapshot, with connect type as FranceConnect 1`] = `
<div>
  <button
    class="fr-connect"
    type="button"
  >
    <span
      class="fr-connect__login"
    >
      any-autheticate-i81n-mock
    </span>
    <span
      class="fr-connect__brand"
    >
      FranceConnect
    </span>
  </button>
</div>
`;

exports[`LoginConnectButton shoud match snapshot, with connect type as ProConnect 1`] = `
<div>
  <button
    class="fr-connect fr-connect--proconnect"
    type="button"
  >
    <span
      class="fr-connect__login"
    >
      any-autheticate-i81n-mock
    </span>
    <span
      class="fr-connect__brand"
    >
      ProConnect
    </span>
  </button>
</div>
`;
